### PR TITLE
[Jetpack Social] Implement UseCase to verify if we should display social sharing limit UI

### LINKED ISSUE #18730: [Jetpack Social] Parse features object in SiteModel

### PR DESCRIPTION
**Pull request by RenanLukas**
Fixes #18730 

**Do not merge until FluxC PR is merged and `trunk` version hash is updated here.**

To test: Nothing to test yet until the `UseCase` starts being used. Please confirm that the condition is correct according to the project specs and that migration from previous DB versions work as correctly.

## Regression Notes
1. Potential unintended areas of impact
None

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests

3. What automated tests I added (or what prevented me from doing so)
`GetIsSocialSharingLimitedUseCaseTest.kt`

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/usecase/social/GetIsSocialSharingLimitedUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.util.extensions.doesNotContain
+import javax.inject.Inject
+
+class GetIsSocialSharingLimitedUseCase @Inject constructor(
+    private val siteStore: SiteStore,
+) {
+    fun execute(siteId: Long): Boolean =
+        siteStore.getSiteBySiteId(siteId)?.run {
+            !isHostedAtWPCom && planActiveFeaturesList?.doesNotContain(FEATURE_SOCIAL_SHARES_1000) ?: false
```
Comment (thomashorta) [inline WordPress/src/main/java/org/wordpress/android/usecase/social/GetIsSocialSharingLimitedUseCase.kt]: 🔍  To make it easier to understand, I would add parenthesis around the second condition, wdyt?

⚠️ Also, I think this fallback is inverted, isn't it? If there's no Features List then I'd argue that site does not contain `FEATURE_SOCIAL_SHARES_1000`, so that should return `true` (similar to [iOS implementation](https://github.com/wordpress-mobile/WordPress-iOS/blob/09143b4ffc4539ead1cbaab944c7af43d5e0c51f/WordPress/Classes/Models/Blog%2BJetpackSocial.swift#L10-L11)), does that make sense?

```suggestion
            !isHostedAtWPCom && (planActiveFeaturesList?.doesNotContain(FEATURE_SOCIAL_SHARES_1000) ?: true)
```

File: WordPress/src/test/java/org/wordpress/android/usecase/social/GetIsSocialSharingLimitedUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        val actual = classToTest.execute(1L)
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `Should return TRUE if site is self hosted AND does not have social-shares-1000 feature active`() {
+        val siteModel = SiteModel().apply {
+            siteId = 1L
+            setIsJetpackInstalled(true)
+            planActiveFeatures = ""
```
Comment (thomashorta) [inline WordPress/src/test/java/org/wordpress/android/usecase/social/GetIsSocialSharingLimitedUseCaseTest.kt:69]: I would also add a test with `planActiveFeatures == null`, to cover all scenarios. See comment above.

**Review by thomashorta — COMMENTED** [review]
Thanks for the changes! 🙇 

I added a couple of comments that might require changes.

File: WordPress/src/main/java/org/wordpress/android/usecase/social/GetIsSocialSharingLimitedUseCase.kt (hunk 1 above)
Comment (RenanLukas) [reply WordPress/src/main/java/org/wordpress/android/usecase/social/GetIsSocialSharingLimitedUseCase.kt]: Good catch! Makes sense, thanks.

**Review by thomashorta — APPROVED** [review]
Thanks for the changes, everything looks good to me! :shipit:
